### ISSUE TITLE
Hermetic pilot v2 tests

### ISSUE THREAD

**Issue by costinm** [istio/istio#3938]
The pilot integration tests are using the same service and rule names for all tests.
For alpha3 ( I don't think it's worth spending time on the alpha1 ) we should create one 
service and one rule name for each test. This will allow tests to run in parallel or be better isolated.

In addition, having the tests isolated allows the setup to deploy a test environment and avoid
having to wait on each test.

Separate issue for creating few dedicated 'dynamic' tests, that explicitly test the 'eventual consistency'.
Current tests are dynamic ( apply rule and test) but they sleep or retry instead of verifying that
during dynamic updates the state is stable. Laurent has a dynamic test which can be used as 
a basis.

**Comment by andraxylia** [comment]
is this still the plan?
